### PR TITLE
Copter: added flowhold, poshold, stabilize, circle, drift and althold modes as auxiliary switch options

### LINKED ISSUE #10929: Copter: add FlowHold to auxiliary switch options
All (or most) flight modes are available as auxiliary switch options but flowHold seems to be missing so we should add it.  It seems we are also missing these modes: Stabilize, AltHold, Circle, Drift, Sport (I wouldn't bother with this one though), PosHold, Guided_NoGPS (I'd leave this one out as well)

### PR DESCRIPTION
**Pull request by arjun7965**
fixes #10929

### CODE REVIEW

**Comment by lvale** [comment]
Although I like the idea of having some specific flight modes on Options like we have today for RTH, I'm not sure if this is the correct path, before we can come up with a solution for several actions per rc channel (ie multiple active ranges per rc channel, right @peterbarker ?) like we only have 3 fixed settings on option channels and 6 on flight mode channel, and in this case what happens when the option is switched off? Returns to the Flight Mode channel selected ?

I see this solution a great first use case for the Scripting system @WickedShell ?

**Comment by peterbarker** [comment]
On Sun, 7 Apr 2019, Luis Vale Gonçalves wrote:

> Although I like the idea of having some specific flight modes on Options like we have today for RTH, I'm not sure if this is the correct path, before we
> can come up with a solution for several actions per rc channel (ie multiple active ranges per rc channel, right @peterbarker ?) like we only have 3 fixed

AFAICS the two are unconnected.

And yes, we re-poll the mode switch when the switch goes "low".

File: libraries/RC_Channel/RC_Channel.h [hunk 1]
```diff
@@ -163,6 +163,12 @@ class RC_Channel {
         GPS_DISABLE  =        65, // disable GPS for testing
         RELAY5 =              66, // Relay5 pin on/off
         RELAY6 =              67, // Relay6 pin on/off
+	STABILIZE =           68, // stabilize mode
```
Comment (muramura) [inline libraries/RC_Channel/RC_Channel.h]: The coding style does not use tab codes.
White space is 4 characters.

**Comment by lvale** [comment]
> AFAICS the two are unconnected.

And yes, we re-poll the mode switch when the switch goes "low".

@peterbarker 
ok, so we have our flight mode switch at Loiter

Now, you use a option switch high to go to AltHold 

Vehicle goes to Alt Hold

Now the same option switch to low -> does it keep alt hold ? Yes AFAICS..... IMHO should revert to the last flight mode...

File: ArduCopter/RC_Channel.cpp [hunk 2]
```diff
@@ -105,6 +105,12 @@ void RC_Channel_Copter::init_aux_function(const aux_func_t ch_option, const aux_
     case AUX_FUNC::ZIGZAG:
     case AUX_FUNC::ZIGZAG_SaveWP:
         break;
+    case AUX_FUNC::STABILIZE:
```
Comment (peterbarker) [inline ArduCopter/RC_Channel.cpp]: Should these go above the break?!

File: libraries/RC_Channel/RC_Channel.h (hunk 1 above)
Comment (arjun7965) [reply libraries/RC_Channel/RC_Channel.h]: will do

File: ArduCopter/RC_Channel.cpp (hunk 2 above)
Comment (arjun7965) [reply ArduCopter/RC_Channel.cpp]: I think the break is for ZIGZAG_SaveWP, maybe?

Comment (peterbarker) [reply ArduCopter/RC_Channel.cpp]: Ah.  I think you might like to read up on the way switch statements work in C/C++ :-)

Please move the break to underneath your lines.

**Comment by peterbarker** [comment]
On Sun, 7 Apr 2019, Luis Vale Gonçalves wrote:

> Now the same option switch to low -> does it keep alt hold ? Yes AFAICS..... IMHO should revert to the last flight mode...

It should move to the mode specified by the mode switch (not necessarily 
the last flight mode...).

Do you have a log showing it behaving differently?

File: ArduCopter/RC_Channel.cpp (hunk 2 above)
Comment (arjun7965) [reply ArduCopter/RC_Channel.cpp]: My bad. Fixed

**Comment by rmackay9** [comment]
Merged with a couple of formatting fixes, thanks!

**Pull request closed without merging**